### ISSUE TITLE
SPM

### ISSUE THREAD

**Issue by bsbhro** [aciidb0mb3r/ankit.im#1]
Hi, Happy New Year!
I got this: " swift-build: Your source structure is not supported due to invalid sources layout: /home/bs/Programme/HelloSwift/SwiftyFactorial. There should be no source files under: /home/bs/Programme/HelloSwift/SwiftyFactorial. "
May be, the strategy of the SwiftPackageManager changed in the past days. I made a folder Sources and moved the main.swift in it. Then it worked.
A question of a beginner: Why do we have a Package.swift in CGetFactorial ? I read it, "This can be left empty but the file has to be present", but when it's used?
Regards, Bernhard
![bildschirmfoto_2016-01-01_13-41-58](https://cloud.githubusercontent.com/assets/2652120/12070888/b8ac0ee2-b08f-11e5-85e0-03282f5a9676.png)

**Comment by aciidb0mb3r** [comment]
Hi,
Are you working with 31st December snapshot? I'll check using that but the [doc](https://github.com/apple/swift-package-manager/blob/master/Documentation/SourceLayouts.md) still says:
```
It is acceptable to have no Sources directory, in which case the root directory is treated as a single module (place your sources there) or sub directories of the root are considered modules. Use this layout convention for simple projects.
```


Package.swift inside CGetFactorial is needed by the SPM to build the package. I guess SPM relies on this file to consider it as a swift package. It can also be used to declare other C dependencies if any as described here : https://github.com/apple/swift-package-manager/blob/master/Documentation/SystemModules.md

**Comment by bsbhro** [comment]
No, it was 'swift-2.2-SNAPSHOT-2015-12-22-a-ubuntu15.10'. But I know since the beginning of Dec. that it worked only with the Sources-folder when using 'swift build'.
Now it's a bit clear: At first I cloned CGetFactorial. The file Package.swift there irritates. Not we need it, but the next it using program :-) I have to 'sudo make'. Without this step we'll get an error. Then I cloned the SwiftyFactorial and changed its Package.swift to '../CGetFactorial' and the 'swift build' worked.
Thank you very much for the blog article and the two linked examples.  Regards
![bildschirmfoto_2016-01-01_16-28-05](https://cloud.githubusercontent.com/assets/2652120/12071343/4aa72c20-b0a7-11e5-8975-df17ae911427.png)
